File: app/src/main/java/com/tanka/accessories/todoroom/data/room/NoteDao.java
package com.tanka.accessories.todoroom.data.room;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import com.tanka.accessories.todoroom.data.model.Note;

import java.util.List;

/**
 * Created by access-tanka on 11/16/17.
 */

@Dao
public interface NoteDao {

    @Insert
    void insertAll(Note... notes);

    @Update
    void updateAll(Note... notes);

    @Query("SELECT * FROM note")
    List<Note> getAllNotes();

    @Query("SELECT * FROM note where title LIKE :keyword")
    List<Note> getNote(String keyword);

    @Delete
    void deleteAll(Note... notes);

    @Update
    void editNote(Note note);

    @Query("UPDATE Note SET `title` = :txTitle, `body` = :txBody, `date` = :txDate, `type` = :txType   WHERE id = :tid")
    void updateNote(long tid, String txTitle, String txBody, String txDate, String txType);
}

File: app/src/main/java/com/tanka/accessories/todoroom/data/room/NoteDao.kt
package com.tanka.accessories.todoroom.data.room


import androidx.room.*
import com.tanka.accessories.todoroom.data.model.Note

/**
 * Created by access-tanka on 11/16/17.
 */
@Dao
interface NoteDao {
    /*
    @Query("SELECT * FROM user")
    suspend fun getAll(): List<User>

    @Insert
    suspend fun insertAll(users: List<User>)

    @Delete
    suspend fun delete(user: User)*/
    @Insert
    fun insertAll(notes: List<Note>)

    @Update
    fun updateAll(vararg notes: Note?)

    @get:Query("SELECT * FROM note")
    val allNotes: List<Note>

    @Query("SELECT * FROM note where title LIKE :keyword")
    fun getNote(keyword: String?): List<Note?>?

    @Delete
    fun deleteAll(vararg notes: Note?)

    @Update
    fun editNote(note: Note?)

    @Query("UPDATE Note SET `title` = :txTitle, `body` = :txBody, `date` = :txDate, `type` = :txType   WHERE id = :tid")
    fun updateNote(tid: Long, txTitle: String?, txBody: String?, txDate: String?, txType: String?)

}